### PR TITLE
Fix/group parameter initial value

### LINKED ISSUE #1730: GroupParameter initial_value cannot be set
Here's a MWE:
```py
from qcodes.instrument.group_parameter import GroupParameter, Group
from qcodes import Instrument

class MyInstrument(Instrument):
    def __init__(self, name, *args, **kwargs):
        super().__init__(name, *args, **kwargs)

        self.add_parameter(name="foo",
                           initial_value=42,
                           parameter_class=GroupParameter
                          )

        self.group = Group([self.foo])

instr = MyInstrument("test")
```

### Expected behaviour
The instrument should have the GroupParameter with the given initial value.

### Actual behaviour
Raises `RuntimeError("Trying to set Group value but no  group defined")`.

### Proposed fix
The `GroupParameter` should defer setting the initial value until it has been added to the group. One way of doing it would be to add something like the following to `GroupParameter.__init__`, before the `super().__init__` call:
```py
if "initial_value" in kwargs:
    self._initial_value = kwargs["initial_value"]
    kwargs["initial_value"] = None
```
and then adding a `GroupParameter.add_to_group` method where the value is actually set, and calling that instead of just setting `parameter.group = self` in `Group.__init__`. I'm not 100% sure if this is the right way to do it.

### System
**qcodes branch**: master

**qcodes commit**: c7eef82d9ab68afb3546fb3c736f2d5b2ff02a14

### PR DESCRIPTION
**Pull request by mgunyho**
Fixes #1730 .

Changes proposed in this pull request:
- Add initial_value kwarg to GroupParameter and appropriate logic for setting initial values for all parameters in a group, after the group has been created.

@astafan8

### CODE REVIEW

File: qcodes/instrument/group_parameter.py [hunk 1]
```diff
@@ -176,6 +181,26 @@ def __init__(self,
         else:
             self.get_parser = self._separator_parser(separator)
 
+        have_initial_values = [p._initial_value is not None
+                               for p in parameters]
+        if any(have_initial_values):
+            if not all(have_initial_values):
+                have_initial_values = [p.name for p in parameters
```
Comment (astafan8) [inline qcodes/instrument/group_parameter.py]: could you please not re-use an existing variable? i suggest `params_with_initial_values` for example.

File: qcodes/tests/test_group_parameter.py [hunk 2]
```diff
@@ -88,3 +98,19 @@ def test_raises_on_get_set_without_group():
...
+    initial_a = 42
+    initial_b = 43
+    dummy = Dummy("dummy", initial_a=initial_a, initial_b=initial_b)
+
+    assert dummy.a() == initial_a
+    assert dummy.b() == initial_b
+
+def test_raise_on_not_all_initial_values():
+
+    expected_err_msg = (r'Either none or all of the parameters in a group '
```
Comment (astafan8) [inline qcodes/tests/test_group_parameter.py]: thanks for a clear error message!

File: qcodes/tests/test_group_parameter.py [hunk 3]
```diff
@@ -66,7 +77,6 @@ def test_sanity():
     assert dummy.a() == 3
     assert dummy.b() == 10
 
-
 def test_raise_on_get_set_cmd():
```
Comment (astafan8) [inline qcodes/tests/test_group_parameter.py]: please, don't remove the space - per pep8 there should be two empty lines between functions.

File: qcodes/tests/test_group_parameter.py [hunk 4]
```diff
@@ -88,3 +98,19 @@ def test_raises_on_get_set_without_group():
     with pytest.raises(RuntimeError) as e:
         param.set(1)
     assert str(e.value) == "('Trying to set Group value but no group defined', 'setting b to 1')"
+
+def test_initial_values():
```
Comment (astafan8) [inline qcodes/tests/test_group_parameter.py]: please, add another empty line between functions (pep8)

File: qcodes/tests/test_group_parameter.py [hunk 5]
```diff
@@ -88,3 +98,19 @@ def test_raises_on_get_set_without_group():
...
+    initial_a = 42
+    initial_b = 43
+    dummy = Dummy("dummy", initial_a=initial_a, initial_b=initial_b)
+
+    assert dummy.a() == initial_a
+    assert dummy.b() == initial_b
+
+def test_raise_on_not_all_initial_values():
```
Comment (astafan8) [inline qcodes/tests/test_group_parameter.py]: please, add another empty line between functions (pep8)

File: qcodes/tests/test_group_parameter.py [hunk 6]
```diff
@@ -88,3 +98,19 @@ def test_raises_on_get_set_without_group():
...
+
```
Comment (astafan8) [inline qcodes/tests/test_group_parameter.py]: and this empty line can be removed :)

**Comment by mgunyho** [comment]
Thanks for the comments! I implemented the fixes you suggested. I don't know why I deleted that one space and didn't adhere to pep8 in general.

**Pull request merged**